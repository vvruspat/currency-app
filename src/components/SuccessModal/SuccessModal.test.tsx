import { render, screen } from "@testing-library/react";
import { SuccessModal } from ".";

test("renders Exchange USD -> EUR", () => {
  render(
    <SuccessModal
      data-testid="test-exchange"
      fromCurrency="USD"
      toCurrency="GBP"
      fromAmount={456.65}
      toAmount={876.45}
    />,
    {}
  );
  const ExchangeElement = screen.getByTestId("test-exchange");

  expect(ExchangeElement).toBeInTheDocument();
  expect(ExchangeElement).toMatchSnapshot();
});
