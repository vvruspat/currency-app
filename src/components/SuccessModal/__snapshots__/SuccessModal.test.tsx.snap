// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Exchange USD -> EUR 1`] = `
<div
  class="modal success-modal"
  data-testid="test-exchange"
>
  <header
    class="modal-header"
  >
    <div
      class="modal-header-content"
    >
      <div
        class="caption"
      >
        <div
          class="title"
        >
          Success!
        </div>
      </div>
    </div>
    <button
      class="icon-button modal-header-close-button"
    >
      <div
        class="icon-button-content"
      >
        <div
          class="icon-button-icon"
        >
          <svg>
            Cross16Icon.svg
          </svg>
        </div>
        <div
          class="icon-button-text"
        >
          Close
        </div>
      </div>
    </button>
  </header>
  <section
    class="modal-content"
  >
    <div
      class="success"
    >
      <div
        class="success-icon"
      >
        <svg>
          Check64Icon.svg
        </svg>
      </div>
      <div
        class="caption success-caption"
      >
        You exchange
      </div>
      <div
        class="title success-body"
      >
        $456.65
         
        to
         
        Â£876.45
      </div>
    </div>
  </section>
</div>
`;
